File: angular/src/app/model/RessourceMaterielle/ordinateur.ts
import {RessourceMaterielle} from './ressourceMaterielle';
import {Salle} from './salle';

export class Ordinateur extends RessourceMaterielle {
  constructor(private _processeur?: string, private _ram?: number, private _disqueDur?: number, private _anneeAchat?: Date,
              private _salles?: Salle[]) {
    super();
  }
  get processeur(): string {
    return this._processeur;
  }

  set processeur(value: string) {
    this._processeur = value;
  }

  get ram(): number {
    return this._ram;
  }

  set ram(value: number) {
    this._ram = value;
  }

  get disqueDur(): number {
    return this._disqueDur;
  }

  set disqueDur(value: number) {
    this._disqueDur = value;
  }

  get anneeAchat(): Date {
    return this._anneeAchat;
  }

  set anneeAchat(value: Date) {
    this._anneeAchat = value;
  }
}

File: java/src/main/java/com/sopra/Factory/model/Ordinateur.java
package com.sopra.Factory.model;

import java.util.Date;
import java.util.Set;

import javax.persistence.Column;
import javax.persistence.DiscriminatorValue;
import javax.persistence.Entity;
import javax.persistence.OneToMany;

import com.fasterxml.jackson.annotation.JsonView;

@Entity
@DiscriminatorValue("ordinateur")
public class Ordinateur extends RessourceMaterielle {
	
	@JsonView(JsonView.class)
	@Column(name = "processeur")
	private String processeur;
	@Column(name = "ram")
	private Integer ram;
	@Column(name = "disque_dur")
	private Integer disqueDur;
	@Column(name = "annee_achat")
	private Date anneeAchat;
	
	@OneToMany(mappedBy="ordinateur")
	//@JsonView(JsonViews.OrdinateurByIdWithStagiaires.class)
    private Set<Stagiaire> stagiaires;
	
	public String getProcesseur() {
		return processeur;
	}
	public void setProcesseur(String processeur) {
		this.processeur = processeur;
	}
	public Integer getRam() {
		return ram;
	}
	public void setRam(Integer ram) {
		this.ram = ram;
	}
	public Integer getDisqueDur() {
		return disqueDur;
	}
	public void setDisqueDur(Integer disqueDur) {
		this.disqueDur = disqueDur;
	}
	public Date getAnneeAchat() {
		return anneeAchat;
	}
	public void setAnneeAchat(Date anneeAchat) {
		this.anneeAchat = anneeAchat;
	}
	
	
	
	public Set<Stagiaire> getStagiaires() {
		return stagiaires;
	}
	public void setStagiaires(Set<Stagiaire> stagiaires) {
		this.stagiaires = stagiaires;
	}
	public Ordinateur() {
		super();
	}
	
	
	

}
